runtime/src/main/java/com/couchbase/quarkus/extension/runtime/CouchbaseRecorder.java

```diff
@@ -20,24 +20,47 @@
 
 import com.couchbase.client.java.Cluster;
 import com.couchbase.client.java.ClusterOptions;
+import com.couchbase.client.java.env.ClusterEnvironment;
 import com.couchbase.client.metrics.micrometer.MicrometerMeter;
 
+import io.micrometer.core.instrument.Meter;
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.core.instrument.config.MeterFilter;
+import io.micrometer.core.instrument.distribution.DistributionStatisticConfig;
 import io.quarkus.runtime.annotations.Recorder;
 
 @Recorder
 public class CouchbaseRecorder {
 
-    public Supplier<Cluster> getCluster(CouchbaseConfig config) {
-        return () -> Cluster.connect(config.connectionString(), config.username(), config.password());
+    public Supplier<Cluster> getCluster(CouchbaseConfig config, boolean metricsEnabled) {
+        var clusterOptions = ClusterOptions.clusterOptions(config.username(), config.password());
+        var clusterEnvironmentBuilder = ClusterEnvironment.builder();
+        if (metricsEnabled) {
+            configureMetrics(clusterEnvironmentBuilder, config.emitInterval());
+        }
+        clusterOptions.environment(clusterEnvironmentBuilder.build());
```

Comment:
This creates a new ClusterEnvironment, and the preferred pattern these days is instead:

```
ClusterOptions.clusterOptions(config.username(), config.password())
   .environment(env -> {
      if (metricsEnabled) {
          env. ... do metrics things here ... ;
      }
   })
```

It leaves the ClusterEnvironment owned by the Cluster, which is a cleaner pattern and ensures it gets cleaned up on the Cluster's lifecycle.